### PR TITLE
Issue 24788

### PR DESCRIPTION
**Pull request by webkul-ratnesh**
### Description (*)

With this PR now you can save "Maximum Qty Allowed in Shopping Cart" & "Minimum Qty Allowed in Shopping Cart"  store wise.
### Fixed Issues (if relevant)

1. magento/magento2#24788 : Issue title

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/CatalogInventory/etc/adminhtml/system.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
                     <backend_model>Magento\CatalogInventory\Model\Config\Backend\Backorders</backend_model>
                     <comment>Changing can take some time due to processing whole catalog.</comment>
                 </field>
-                <field id="max_sale_qty" translate="label" type="text" sortOrder="4" showInDefault="1" showInWebsite="0" showInStore="0" canRestore="1">
+                <field id="max_sale_qty" translate="label" type="text" sortOrder="4" showInDefault="1" showInWebsite="0" showInStore="1" canRestore="1">
```
Comment (dmytro-ch) [inline app/code/Magento/CatalogInventory/etc/adminhtml/system.xml]:
```suggestion
                <field id="max_sale_qty" translate="label" type="text" sortOrder="4" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
```

File: app/code/Magento/CatalogInventory/etc/adminhtml/system.xml [hunk 2]
```diff
@@ -62,7 +62,7 @@
                     <validate>validate-number</validate>
                     <backend_model>Magento\CatalogInventory\Model\System\Config\Backend\Minqty</backend_model>
                 </field>
-                <field id="min_sale_qty" translate="label" sortOrder="6" showInDefault="1" showInWebsite="0" showInStore="0" canRestore="1">
+                <field id="min_sale_qty" translate="label" sortOrder="6" showInDefault="1" showInWebsite="0" showInStore="1" canRestore="1">
```
Comment (dmytro-ch) [inline app/code/Magento/CatalogInventory/etc/adminhtml/system.xml]:
```suggestion
                <field id="min_sale_qty" translate="label" sortOrder="6" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
As for me, It's a quite disputable case when you need to set the Sale Quantity restrictions on Store View level. Nonetheless, according to the implementation of `\Magento\CatalogInventory\Model\Configuration`, it can be possible.
The only thing I would improve is to allow setting these values on the website level as well.

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-6130](https://jira.corp.magento.com/browse/ENGCOM-6130) has been created to process this Pull Request

**Comment by sdzhepa** [comment]
Hello @webkul-ratnesh 

Thank you for contribution and collaboration!

Related issue https://github.com/magento/magento2/issues/24788 has been closed. 
Please see more details in the initial issue https://github.com/magento/magento2/issues/24788#issuecomment-545567456

At this moment work on this global solution is in progress in the scope of this task https://github.com/magento/inventory/pull/2297 . All planned work and design is described in this document [Stock and Source Configuration design](https://github.com/magento/inventory/wiki/Stock-and-Source-Configuration-design)

**Comment by alevashov** [comment]
As I see in Magento ver. 2.4.5-p1 this solution is still not being implemented, which is a shame, because it is needed for the cases of running B2C and B2B stores from the same Magento instance

**Pull request closed without merging**